kernel/src/client/arrow_utils.rs

```diff
@@ -47,26 +68,32 @@ pub(crate) fn generate_mask(
     }
 }
 
-/// Reorder a RecordBatch to match `requested_schema`. This method takes `indicies` as computed by
-/// [`get_requested_indicies`] as an optimization. If the indicies are in order, then we don't need
-/// to do any re-ordering.
+/// Reorder a RecordBatch to match `requested_ordering`. This method takes `mask_indicies` as
+/// computed by [`get_requested_indicies`] as an optimization. If the indicies are in order, then we
+/// don't need to do any re-ordering. Otherwise, for each non-zero value in `requested_ordering`,
+/// the column at that index will be added in order to returned batch
 pub(crate) fn reorder_record_batch(
-    requested_schema: Arc<ArrowSchema>,
     input_data: RecordBatch,
-    indicies: &[usize],
+    mask_indicies: &[usize],
+    requested_ordering: &[usize],
 ) -> DeltaResult<RecordBatch> {
-    if indicies.windows(2).all(|is| is[0] <= is[1]) {
+    if mask_indicies.windows(2).all(|is| is[0] <= is[1]) {
         // indicies is already sorted, meaning we requested in the order that the columns were
         // stored in the parquet
```

Comment:
this does just check that i think? it just checks that for each pair of two elements the earlier one is `<=` to the later one. so for your example it would be checking that `2<=4`, which is `true`.

this can be changed to `<` probably, since we won't have duplicate indices.

Reply:
Ah, good point. Not sure what I was thinking.